Analysis/Smells/AstSmells/LongChainingOfDotFunctions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Esprima;
+using Esprima.Ast;
+
+namespace BSDetector.Analysis.Smells.AstSmells
+{
+    public class LongChainingOfDotFunctions : AstSmell
+    {
+        public override string SmellName => "CHAIN_DOT_FUNC";
+
+        public override void AnalyzeNode(INode node, int chainLength)
+        {
+            if (node is MemberExpression MemberNode)
+            {
+                chainLength += 1;
+                foreach (var child in node.ChildNodes) AnalyzeNode(child, chainLength, MemberNode.Location);
+            }
+            else
+            {
+                foreach (var child in node.ChildNodes) AnalyzeNode(child, chainLength);
+            }
```

Comment:
I'm not sure, but you can try to reduce lines of code by using ++chainLength as a parameter for AnalyzeNode.